src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -675,10 +677,35 @@ class WordHighlighter {
 			// 		1) we have text focus, and a valid query was updated.
 			// 		2) we do not have text focus, and a valid query is cached.
 			// the query will ALWAYS have the correct data for the current highlight request, so it can always be passed to the workerRequest safely
-			if (!WordHighlighter.query || !WordHighlighter.query.modelInfo || WordHighlighter.query.modelInfo.model.isDisposed()) {
+			if (!WordHighlighter.query || !WordHighlighter.query.modelInfo) {
 				return;
 			}
-			this.workerRequest = this.computeWithModel(WordHighlighter.query.modelInfo.model, WordHighlighter.query.modelInfo.selection, WordHighlighter.query.word, otherModelsToHighlight);
+			const queryModelRef = await this.textModelService.createModelReference(WordHighlighter.query.modelInfo.modelURI);
```

Comment:
Why is this needed and where is this reference disposed?

Reply:
> Why is this needed

answered in #230635, let me know if you'd like to discuss this problem further.

> where is this reference disposed

to no surprise, given the leak, it wasn't 😞  See #230733 for the fix